src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -0,0 +1,580 @@
+        }
+    }
+
+    private Optional<Duration> getForwardPathLatency(InputData inputData) {
+        return getPathLatency(inputData, s ->
+                getIslByEndPoints(s.getSrcSwitchId(), s.getSrcPort(), s.getDestSwitchId(), s.getDestPort()));
+    }
+
+    private Optional<Duration> getReversePathLatency(InputData inputData) {
+        return getPathLatency(inputData, s ->
+                getIslByEndPoints(s.getDestSwitchId(), s.getDestPort(), s.getSrcSwitchId(), s.getSrcPort()));
+    }
+
+    private String getForwardLatencyErrorMessage(InputData data, Duration expectedLatency, String latencyType,
+                                                 Duration actualLatency) {
+        return String.format(
+                "Requested %s is too low on the path between: switch %s and switch %s. "
+                        + "Requested %d ms, but the sum on the path is %d ms.",
+                latencyType,
+                data.getPath().getSrcSwitchId(), data.getPath().getDestSwitchId(),
+                expectedLatency.toMillis(), actualLatency.toMillis());
+    }
+
+    private String getReverseLatencyErrorMessage(InputData data, Duration expectedLatency, String latencyType,
+                                                 Duration actualLatency) {
+        return String.format(
+                "Requested %s is too low on the path between: switch %s and switch %s. "
+                        + "Requested %d ms, but the sum on the path is %d ms.",
+                latencyType,
+                data.getPath().getDestSwitchId(), data.getPath().getSrcSwitchId(),
+                expectedLatency.toMillis(), actualLatency.toMillis());
+    }
+
+    private String getForwardBandwidthErrorMessage(InputData data, long actualBandwidth) {
+        return String.format(
+                "There is not enough bandwidth between end points: switch %s port %d and switch %s port %d"
+                        + " (forward path). Requested bandwidth %d, but the link supports %d.",
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort(),
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort(),
+                data.getPath().getBandwidth(), actualBandwidth);
+    }
+
+    private String getReverseBandwidthErrorMessage(InputData data, long actualBandwidth) {
+        return String.format(
+                "There is not enough bandwidth between end points: switch %s port %d and switch %s port %d"
+                        + " (reverse path). Requested bandwidth %d, but the link supports %d.",
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort(),
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort(),
+                data.getPath().getBandwidth(), actualBandwidth);
+    }
+
+    private String getNoLinkOnPath() {
+        return "Path latency cannot be calculated because there is no link at least at one path segment.";
```

Comment:
I think you should specify which exactly link is missed. It will look like "Something went wrong" otherwise.

Reply:
I split the validation into per-segment and per-path. Per-path doesn't (shouldn't) look at the segments level and this is why there is no a specific segment in the message. It only says that per-path validation cannot be performed. 
The diagnostic information about specific segments is there, but on per-segment level. In case there are links missing, per-segment validation would report "there is no ISL here and here", while per-path validations report that it is not possible to execute them.